src/helpers/helpers.canvas.js

```diff
@@ -291,3 +291,78 @@ export function _bezierCurveTo(ctx, previous, target, flip) {
 		target.x,
 		target.y);
 }
+
+/**
+ * Render text onto the canvas
+ */
+export function renderText(ctx, text, x, y, font, opts = {}) {
+	const lines = isArray(text) ? text : [text];
+	let i, line;
+
+	ctx.save();
+
+	if (opts.translation) {
+		ctx.translate(opts.translation[0], opts.translation[1]);
+	}
+
+	if (!isNullOrUndef(opts.rotation)) {
+		ctx.rotate(opts.rotation);
+	}
+
+	ctx.font = font.fontString;
+
+	if (opts.color) {
+		ctx.fillStyle = opts.color;
+	}
+
+	if (opts.textAlign) {
+		ctx.textAlign = opts.textAlign;
+	}
+
+	if (opts.textBaseline) {
+		ctx.textBaseline = opts.textBaseline;
+	}
+
+	for (i = 0; i < lines.length; ++i) {
+		line = lines[i];
+
+		if (opts.stroke) {
+			if (opts.strokeColor) {
+				ctx.strokeStyle = opts.strokeColor;
+			}
+
+			if (!isNullOrUndef(opts.strokeWidth)) {
+				ctx.lineWidth = opts.strokeWidth;
+			}
+
+			ctx.strokeText(line, x, y, opts.maxWidth);
+		}
+
+		ctx.fillText(line, x, y, opts.maxWidth);
+
+		if (opts.strikethrough || opts.underline) {
+			/**
+			 * Now that IE11 support has been dropped, we can use more
+			 * of the TextMetrics object. The actual bounding boxes
+			 * are unflagged in Chrome, Firefox, Edge, and Safari so they
+			 * can be safely used.
+			 * See https://developer.mozilla.org/en-US/docs/Web/API/TextMetrics#Browser_compatibility
+			 */
+			const metrics = ctx.measureText(line);
+			const left = x - metrics.actualBoundingBoxLeft;
+			const right = x + metrics.actualBoundingBoxRight;
+			const top = y - metrics.actualBoundingBoxAscent;
+			const bottom = y + metrics.actualBoundingBoxDescent;
+			const yDecoration = opts.strikethrough ? (top + bottom) / 2 : bottom;
+
+			ctx.beginPath();
+			ctx.lineWidth = opts.decorationWidth || 2;
+			ctx.moveTo(left, yDecoration);
+			ctx.lineTo(right, yDecoration);
+			ctx.stroke();
```

Comment:
I think we need to make sure correct color is used here (with or without stroke).

Reply:
You're right. We need to set the `strokeStyle` to the fill colour after the text is stroked